src/compiler/parser.rs

```diff
@@ -455,6 +454,9 @@ impl ParserType {
     fn parse_primary_expression(&mut self) -> Expression {
         match self.token() {
             SyntaxKind::NumericLiteral => return self.parse_literal_node().into(),
+            SyntaxKind::TrueKeyword | SyntaxKind::FalseKeyword => {
+                return self.parse_token_node().into()
```

Comment:
...and here this is calling `BaseNode::into::<Expression>()` (which wraps in an `Expression::TokenExpression`)